### ISSUE TITLE
Future warning when testing `if isinstance(x, Viewer)`

### ISSUE THREAD

**Issue by haesleinhuepf** [napari/napari#4745]
## 🐛 Bug

When testing if a variable is a Viewer, I receive a Future warning that this is no longer possible in napari 0.5.0

## To Reproduce

1. Install [devbio-napari](https://github.com/haesleinhuepf/devbio-napari)
2. Open a 4D dataset of shape `[1, d, h, w]`
3. Apply the menu `Tools > Filtering / Noise Removal > Gaussian (n-SimpleITK)`

This warning pops up:
```
c:\structure\code\napari-workflows\src\napari_workflows\_workflow.py:265: FutureWarning: Private attribute access ('Viewer.__fields__') in this context (e.g. inside a plugin widget or dock widget) is deprecated and will be unavailable in version 0.5.0!
```

The mentioned line can be found [here](https://github.com/haesleinhuepf/napari-workflows/blob/c62dfefabeb0ae6e00519d2295c325e1a9fa97d7/src/napari_workflows/_workflow.py#L265):
```
if isinstance(args[-1], Viewer):
```

## Expected behavior

Testing if a variable is of type Viewer should not cause any warnings. It should also be possible to do this napari 0.5.0

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

```
napari: 0.4.15
Platform: Windows-10-10.0.19044-SP0
Python: 3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:50:36) [MSC v.1929 64 bit (AMD64)]
Qt: 5.15.3
PyQt5: 5.15.4
NumPy: 1.22.4
SciPy: 1.8.1
Dask: 2022.6.1
VisPy: 0.9.6

OpenGL:
- GL version: 4.6.0 - Build 30.0.101.1029
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1920x1200, scale 1.0

Plugins:
- PlatyMatch: 0.0.3
- RedLionfish: 0.5
- animation: 0.0.2
- clEsperanto: 0.18.0
- console: 0.0.4
- devbio_napari: 0.5.7
- napari-3d-ortho-viewer: 0.0.1
- napari-accelerated-pixel-and-object-classification: 0.8.1
- napari-assistant: 0.3.8
- napari-brightness-contrast: 0.1.7
- napari-clusters-plotter: 0.5.1
- napari-crop: 0.1.5
- napari-curtain: 0.1.1
- napari-dexp2: 0.0.8.dev0+g4c92044.d20220627
- napari-folder-browser: 0.1.3
- napari-layer-details-display: 0.1.3
- napari-mouse-controls: 0.1.3
- napari-plot-profile: 0.2.1
- napari-plugin-search: 0.1.3
- napari-segment-blobs-and-things-with-membranes: 0.2.22
- napari-simpleitk-image-processing: 0.3.1
- napari-tabu: 0.1.5
- napari-time-slicer: 0.4.7
- napari-tools-menu: 0.1.15
- napari-workflow-inspector: 0.2.2
- napari-workflow-optimizer: 0.1.4
- napari_skimage_regionprops1: 0.5.2
- napari_skimage_regionprops2: 0.5.2
- scikit-image: 0.4.15
- svg: 0.1.6
- the-segmentation-game: 0.1.0
```

 - Any other relevant information:

**Comment by Czaki** [comment]
the problem is here https://github.com/napari/napari/blob/8beccad1aee695dc90dd2d164c3e90211e9e80a8/napari/utils/_proxies.py#L46

this clause should be extended to not catch names in the format `__something__`

**Comment by lucyleeow** [comment]
I think this has been solved by #4792

**Comment by haesleinhuepf** [comment]
Awesome, big thanks @lucyleeow !
